src/Components/Components/src/NavigationManager.cs

```diff
@@ -27,6 +30,7 @@ public abstract class NavigationManager
                 _locationChanged -= value;
             }
         }
+        public Func<string, List<string>>? OnNavigate { get; set; }
```

Comment:
Should these return `AssemblyName` instances?

Reply:
Good call!

Reply:
Eh -- actually this ends up being a little unnecessary since the list of assemblies needs to be mapped to `string[]` for the JS interop call and it's likely that the name of the assembly is the only field that we will use from the AssemblyName class.